torchmetrics/metric.py

```diff
@@ -453,7 +455,9 @@ def __getitem__(self, idx):
 
     @property
     def is_differentiable(self):
-        raise NotImplementedError
+        # There is a bug in PyTorch that leads to properties being executed during scripting
+        # To make the metric scriptable, we add property to ignore list and switch to return None here
+        return None
```

Comment:
that was my concern earlier... https://github.com/PyTorchLightning/metrics/pull/154#discussion_r612483225

Reply:
It should hopefully be OK to throw once PyTorch fixes the issue (it's not respecting the ignore on properties!), but that would be a while till it gets to the released version, etc.